Rust/src/random_cut_forest.rs

```diff
@@ -333,27 +392,67 @@ mod tests {
     #[test]
     fn gaussian_blob() {
         let num_points = 1000;
-        let dimension = 3;
-        let mut forest: RandomCutForest<f32> = RandomCutForestBuilder::new(dimension).build();
+        let dimensions = 3;
+        let mut forest: RandomCutForest<f32> = RandomCutForestBuilder::new(dimensions).build();
 
-        let points = randn(num_points, dimension);
+        let points = randn(num_points, dimensions);
         for point in points.iter() {
             forest.update(point.clone());
         }
+        let mut anomaly = vec![0.0; dimensions];
+        anomaly[0] = 5.0;
 
+        //
+        // test anomaly score
+        //
         let scores: Vec<f32> = points.iter().map(|p| forest.anomaly_score(&p)).collect();
         let scores_mean: f32 = scores.iter().sum::<f32>() / num_points as f32;
         let scores_max: f32 = scores.iter().fold(0.0, |max_s, s| f32::max(max_s, *s));
+        let anomaly_score = forest.anomaly_score(&anomaly);
+
+        println!("[gaussian_blob] scores mean     = {}", scores_mean);
+        println!("[gaussian_blob] scores max      = {}", scores_max);
+        println!("[gaussian_blob] anomalous score = {}", anomaly_score);
+        assert!(anomaly_score > scores_mean);
+        assert!(anomaly_score > scores_max);
+
+        //
+        // test attrribution score
+        //
+        let mut attribution_one_mean: f32 = 0.0;
+        let anomaly_attribution = forest.anomaly_attribution(&anomaly);
+        let anomaly_sum = anomaly_attribution.total_hi_lo_sum();
+        for i in 0..num_points {
+            let attribution = forest.anomaly_attribution(&points[i]);
+            let relative_error = (attribution.total_hi_lo_sum() - scores[i]).abs() / scores[i];
+
+            attribution_one_mean += attribution.hi_lo_sum(0);
+            assert!(relative_error < 0.01);
+            assert!(anomaly_attribution.hi_lo_sum(0) > attribution.hi_lo_sum(0));
```

Comment:
For each point in the forest we test that the sum of it's attribution scores (`total_hi_lo_sum()`) is equal to its anomaly score. To do this "robustly", we check if the relative error between the attribution sum and the score is close to zero; specifically, within 1% relative error.

Reply:
@cswiercz TODO - because we're traversing the same cuts we should test if the error is close to machine epsilon.